### PR TITLE
Feat / Authorization Module

### PR DESCRIPTION
**Pull request by nahuelRosas**
## Summary

This update introduces significant enhancements to the authorization module, including new interfaces, classes, enums, and types for improved authorization handling.

### Modifications

1. **Introduced `ICaslAbilityFactory` Interface**:
   - Creates `CaslAbility` objects for users to manage their permissions.

2. **Added `IPolicyHandler` Interface**:
   - Defines a common interface for handling authorization policies.

3. **Implemented `BasePolicyHandler` Class**:
   - An abstract class for handling authorization policies, checking user permissions based on actions and subjects.

4. **Created `AuthorizationService` Class**

5. **Developed `CaslAbilityFactory` Class**
 
6. **Added `PoliciesGuard` Class**:
   - A guard that applies authorization policies to routes, using policy handlers.

7. **Defined `AppAction` Enum**:
   - Enumerates actions such as manage, read, create, update, and delete.

8. **Defined `AppRole` Enum**:
   - Enumerates user roles including admin and regular.

9. **Created `AppAbility` Type**:
    - Represents abilities with actions and subjects.

10. **Created `AppSubjects` Type**:
    - Defines subjects within the application, including the `User` class.

11. **Added `DefinePermissions` Type**:
    - Defines permissions based on user roles.

12. **Implemented `AuthorizationModule`**:
    - A module for managing authorization logic, including providers for the authorization service, CASL ability factory, and policy handling.

### Changes Overview

- **Lines Added**: 332
- **Lines Deleted**: 0

**File Changes:**
- `src/modules/authorization/application/interface/policy_handler.interface.ts`: 5 lines added
- `src/modules/authorization/application/policy/base_policy.handler.ts`: 40 lines added
- `src/modules/authorization/application/service/authorization.service.ts`: 44 lines added
- `src/modules/authorization/authorization.module.ts`: 70 lines added
- `src/modules/authorization/domain/app-action.enum.ts`: 7 lines added
- `src/modules/authorization/domain/role.enum.ts`: 4 lines added
- `src/modules/authorization/infraestructure/casl/factory/casl-ability.factory.ts`: 37 lines added
- `src/modules/authorization/infraestructure/casl/type/app-ability.type.ts`: 9 lines added
- `src/modules/authorization/infraestructure/casl/type/app-subjects.type.ts`: 8 lines added
- `src/modules/authorization/infraestructure/policy/decorator/policy.decorator.ts`: 8 lines added
- `src/modules/authorization/infraestructure/policy/guard/policy.guard.ts`: 77 lines added
- `src/modules/authorization/infraestructure/policy/storage/policies-handler.storage.ts`: 22 lines added
- `src/modules/authorization/infraestructure/policy/type/define-permissions.type.ts`: 10 lines added
- `src/modules/authorization/infraestructure/policy/type/permissions-definition.interface.ts`: 7 lines added

### CODE REVIEW

File: src/modules/authorization/application/interface/authorization.service.interface.ts
Comment (anarod13) [inline src/modules/authorization/application/interface/authorization.service.interface.ts:1]: Same as last pr

File: src/modules/authorization/authorization.module.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  static forFeature(
+    options: IAuthorizationModuleForFeatureOptions,
+  ): DynamicModule {
+    return {
+      module: AuthorizationModule,
+      imports: [CommonModule],
+      providers: [
+        {
+          provide: AUTHORIZATION_SERVICE,
```
Comment (anarod13) [inline src/modules/authorization/authorization.module.ts:40]: Update also here please

**Review by anarod13 — APPROVED** [review]

File: src/modules/authorization/application/interface/authorization.service.interface.ts
Comment (nahuelRosas) [reply src/modules/authorization/application/interface/authorization.service.interface.ts:1]: fixed

File: src/modules/authorization/authorization.module.ts (hunk 1 above)
Comment (nahuelRosas) [reply src/modules/authorization/authorization.module.ts]: fixed

**Pull request merged**